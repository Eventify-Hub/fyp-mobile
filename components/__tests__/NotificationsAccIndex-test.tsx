// __tests__/NotificationsAccIndex.test.tsx

import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import NotificationsAccIndex from '../notificationacc/NotificationAccIndex'; // Adjust if path is different

// ✅ Mock useRouter from expo-router
jest.mock('expo-router', () => ({
  useRouter: () => ({
    back: jest.fn(),
    push: jest.fn(),
  }),
}));

// ✅ Mock getSecureData used in BottomNavigationFinal
jest.mock('@/store', () => ({
  getSecureData: jest.fn((key) => {
    if (key === 'user') {
      return Promise.resolve(
        JSON.stringify({
          _id: 'u001',
          role: 'vendor',
          name: 'Ali Test',
          email: 'ali@example.com',
          phoneNumber: '1234567890',
          address: 'Lahore',
        })
      );
    }
    return Promise.resolve(null);
  }),
}));

describe('NotificationsAccIndex', () => {
  it('shows the header title', () => {
    const { getAllByText } = render(<NotificationsAccIndex />);
    expect(getAllByText('Notifications').length).toBeGreaterThan(0);
  });

  it('renders RSVP switch with default ON', () => {
    const { getByTestId } = render(<NotificationsAccIndex />);
    expect(getByTestId('rsvp-switch').props.value).toBe(true);
  });

  it('renders Activity switch with default OFF', () => {
    const { getByTestId } = render(<NotificationsAccIndex />);
    expect(getByTestId('activity-switch').props.value).toBe(false);
  });

  it('renders Deadline switch with default ON', () => {
    const { getByTestId } = render(<NotificationsAccIndex />);
    expect(getByTestId('deadline-switch').props.value).toBe(true);
  });

  it('toggles RSVP switch from ON to OFF', () => {
    const { getByTestId } = render(<NotificationsAccIndex />);
    const toggle = getByTestId('rsvp-switch');
    fireEvent(toggle, 'valueChange', false);
    expect(toggle.props.value).toBe(false);
  });

  it('toggles Activity switch from OFF to ON', () => {
    const { getByTestId } = render(<NotificationsAccIndex />);
    const toggle = getByTestId('activity-switch');
    fireEvent(toggle, 'valueChange', true);
    expect(toggle.props.value).toBe(true);
  });

  it('toggles Deadline switch from ON to OFF', () => {
    const { getByTestId } = render(<NotificationsAccIndex />);
    const toggle = getByTestId('deadline-switch');
    fireEvent(toggle, 'valueChange', false);
    expect(toggle.props.value).toBe(false);
  });

  it('shows RSVP label text', () => {
    const { getByText } = render(<NotificationsAccIndex />);
    expect(
      getByText('Receive notification when guests RSVP to the event')
    ).toBeTruthy();
  });

  it('shows Activity label text', () => {
    const { getByText } = render(<NotificationsAccIndex />);
    expect(
      getByText('Receive notification for activity that involves me')
    ).toBeTruthy();
  });

  it('shows Activity description text', () => {
    const { getByText } = render(<NotificationsAccIndex />);
    expect(
      getByText('When a guest replies to me, mentions, or tags me')
    ).toBeTruthy();
  });

  it('shows Deadline label text', () => {
    const { getByText } = render(<NotificationsAccIndex />);
    expect(
      getByText('Receive notification about approaching deadlines for tasks')
    ).toBeTruthy();
  });

  it('renders back button and it works', () => {
    const { getByText } = render(<NotificationsAccIndex />);
    const back = getByText('< Back');
    fireEvent.press(back);
    expect(back).toBeTruthy();
  });

  it('renders all switches correctly', () => {
    const { getByTestId } = render(<NotificationsAccIndex />);
    expect(getByTestId('rsvp-switch')).toBeTruthy();
    expect(getByTestId('activity-switch')).toBeTruthy();
    expect(getByTestId('deadline-switch')).toBeTruthy();
  });

  it('renders SafeAreaView container', () => {
    const { UNSAFE_getByType } = render(<NotificationsAccIndex />);
    expect(
      UNSAFE_getByType(require('react-native').SafeAreaView)
    ).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(<NotificationsAccIndex />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
