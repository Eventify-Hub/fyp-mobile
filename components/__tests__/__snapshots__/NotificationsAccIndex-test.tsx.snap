// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsAccIndex matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F8E9F0",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "#E0E0E0",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 15,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#780C60",
            "fontSize": 16,
          }
        }
      >
        &lt;
         Back
      </Text>
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Notifications
    </Text>
    <View
      style={
        {
          "width": 40,
        }
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
    keyboardShouldPersistTaps="handled"
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "borderBottomColor": "#E0E0E0",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 15,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "marginRight": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
              }
            }
          >
            Receive notification when guests RSVP to the event
          </Text>
        </View>
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#780C60"
          style={
            {
              "height": 31,
              "width": 51,
            }
          }
          testID="rsvp-switch"
          thumbTintColor="#FFFFFF"
          tintColor="#E0E0E0"
          value={true}
        />
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "borderBottomColor": "#E0E0E0",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 15,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "marginRight": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
              }
            }
          >
            Receive notification for activity that involves me
          </Text>
          <Text
            style={
              {
                "color": "#848484",
                "fontSize": 14,
                "lineHeight": 20,
                "marginTop": 5,
              }
            }
          >
            When a guest replies to me, mentions, or tags me
          </Text>
        </View>
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#780C60"
          style={
            {
              "height": 31,
              "width": 51,
            }
          }
          testID="activity-switch"
          thumbTintColor="#f4f3f4"
          tintColor="#E0E0E0"
          value={false}
        />
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "borderBottomColor": "#E0E0E0",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 15,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "marginRight": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
              }
            }
          >
            Receive notification about approaching deadlines for tasks
          </Text>
        </View>
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#780C60"
          style={
            {
              "height": 31,
              "width": 51,
            }
          }
          testID="deadline-switch"
          thumbTintColor="#FFFFFF"
          tintColor="#E0E0E0"
          value={true}
        />
      </View>
      <View
        style={
          {
            "height": 80,
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopColor": "#E0E0E0",
        "borderTopWidth": 1,
        "bottom": 0,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "space-around",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#780C60",
            "borderRadius": 25,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "width": 30,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/images/myevent.png",
            }
          }
          style={
            {
              "height": 37,
              "marginBottom": 5,
              "width": 37,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 10,
          }
        }
      >
        My Events
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#780C60",
            "borderRadius": 25,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "width": 30,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://cdn.builder.io/api/v1/image/assets/TEMP/549e73c4-da91-40a5-a5c8-fd173b0e2a62?placeholderIfAbsent=true&apiKey=0a92af3bc6e24da3a9ef8b1ae693931a",
            }
          }
          style={
            {
              "height": 37,
              "marginBottom": 5,
              "width": 37,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 10,
          }
        }
      >
        Messages
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
          "transform": [
            {
              "translateY": -10,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#780C60",
            "borderRadius": 27.5,
            "height": 55,
            "justifyContent": "center",
            "marginBottom": 5,
            "width": 55,
          }
        }
      />
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 10,
          }
        }
      >
        Home
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#780C60",
            "borderRadius": 25,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "width": 30,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://cdn.builder.io/api/v1/image/assets/TEMP/198f4cc8-49ff-4ccc-b97b-619e572143d4?placeholderIfAbsent=true&apiKey=0a92af3bc6e24da3a9ef8b1ae693931a",
            }
          }
          style={
            {
              "height": 37,
              "marginBottom": 5,
              "width": 37,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 10,
          }
        }
      >
        Notifications
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#780C60",
            "borderRadius": 25,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "width": 30,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://cdn.builder.io/api/v1/image/assets/TEMP/73089a6f-a9a6-4c94-9fd1-4cdd5923a137?placeholderIfAbsent=true&apiKey=0a92af3bc6e24da3a9ef8b1ae693931a",
            }
          }
          style={
            {
              "height": 37,
              "marginBottom": 5,
              "width": 37,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 10,
          }
        }
      >
        Account
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
